File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class CategoryCount(BaseModel):
    name: str
    value: int


class TrendPoint(BaseModel):
    week: str
    reports: int


class ResolutionStats(BaseModel):
    total_issues: int
    resolved_issues: int
    resolution_rate: float
    average_resolution_days: float


class HealthCategoryOut(BaseModel):
    label: str
    score: int


class HealthResponse(BaseModel):
    city_health_score: int
    categories: list[HealthCategoryOut]
    methodology: str = (
        "Heuristic: 100 minus a weighted penalty per unresolved issue in the "
        "category (critical=-14, high=-8, medium=-4, low=-1), floored at 0. "
        "This is a project scoring heuristic, not a validated infrastructure "
        "index."
    )


class AuthorityDashboardOut(BaseModel):
    total_issues: int
    critical_issues: int
    pending_issues: int
    resolved_issues: int
    average_resolution_days: float
    infrastructure_health: int

File: frontend/lib/api/analytics.ts
import { apiFetch } from "./client";
import type { HealthCategory } from "@/lib/types";

export interface CategoryCount {
  name: string;
  value: number;
}

export interface TrendPoint {
  week: string;
  reports: number;
}

export interface ResolutionStats {
  total_issues: number;
  resolved_issues: number;
  resolution_rate: number;
  average_resolution_days: number;
}

export interface HealthResponse {
  city_health_score: number;
  categories: HealthCategory[];
  methodology: string;
}

export async function getCategoryBreakdown(): Promise<CategoryCount[]> {
  return apiFetch<CategoryCount[]>("/analytics/categories");
}

export async function getTrends(): Promise<TrendPoint[]> {
  return apiFetch<TrendPoint[]>("/analytics/trends");
}

export async function getResolutionStats(): Promise<ResolutionStats> {
  return apiFetch<ResolutionStats>("/analytics/resolution");
}

export async function getHealth(): Promise<HealthResponse> {
  return apiFetch<HealthResponse>("/analytics/health");
}
